lib/manager/github-actions/extract.ts

```diff
@@ -7,24 +8,42 @@ export function extractPackageFile(content: string): PackageFile | null {
   logger.debug('github-actions.extractPackageFile()');
   const deps: PackageDependency[] = [];
   for (const line of content.split('\n')) {
-    // old github actions syntax will be deprecated on September 30, 2019
-    // after that, the first line can be removed
-    const match =
-      /^\s+uses = "docker:\/\/([^"]+)"\s*$/.exec(line) ||
-      /^\s+uses: docker:\/\/([^"]+)\s*$/.exec(line);
-    if (match) {
-      const [, currentFrom] = match;
+    if (line.trim().startsWith('#')) {
+      continue; // eslint-disable-line no-continue
+    }
+
+    const dockerMatch = /^\s+uses: docker:\/\/([^"]+)\s*$/.exec(line);
+    if (dockerMatch) {
+      const [, currentFrom] = dockerMatch;
       const dep = getDep(currentFrom);
       logger.debug(
         {
           depName: dep.depName,
           currentValue: dep.currentValue,
           currentDigest: dep.currentDigest,
         },
-        'Docker image inside GitHub Actions'
+        'Docker image inside GitHub Workflow'
       );
+      dep.depType = 'docker';
       dep.versioning = dockerVersioning.id;
       deps.push(dep);
+      continue; // eslint-disable-line no-continue
+    }
+
+    const tagMatch = /^\s+-?\s+?uses: (?<depName>[\w-]+\/[\w-]+)(?<path>.*)?@(?<currentValue>.+?)\s*?$/.exec(
+      line
+    );
+    if (tagMatch?.groups) {
+      const { depName, currentValue } = tagMatch.groups;
+      const dep: PackageDependency = {
+        depName,
+        currentValue,
+        commitMessageTopic: '{{depName}}} action',
+        datasource: githubReleasesDatasource.id,
```

Comment:
What was the reason to change this? Can't someone specify a tag in a workflow and it works? If so then we should look up tags? If GitHub enforces that you must *release* the tag then this is correct, but otherwise we can end up with a scenario where someone specifies a valid tag but we don't support it because we only look up releases.

Reply:
It's a weird one. For a version to show up the marketplace, it must first be released and published. However, you're right that technically any tag can be referenced still (and GitHub uses this to tag their latest release with both the vMajor and vMajor.Minor.Patch tags).

I'm fine either way. Both github-tags and github-releases are applicable, just leaning toward the more stringent behavior.

Reply:
I think let's go with tags then, as otherwise I can already predict future issues of people asking why Renovate isn't upgrading their Actions :)

Reply:
Switched back to tags :+1: